### ISSUE TITLE
Create user facing documentation

### ISSUE THREAD

**Issue by javerbukh** [spacetelescope/specviz#610]
- Video tutorials
- Read through documentation and decide what needs to be done

**Comment by javerbukh** [comment]
Includes finding scientists and user testing the documentation. This would be for quality assurance.

**Comment by javerbukh** [comment]
@javerbukh will help to break this issue up and @kassin will help to find scientists. @javerbukh and @brechmos-stsci will be working on the video tutorial.

**Comment by robelgeda** [comment]
My review of the SpecViz docs [Stable] :
-	SpecViz logo needs to be larger on the side bar.
-	Installation page looks very good:
	-	Can SpecViz run on python 2.x?
-	The subsections in the Getting Started section should be their own sections:
	-	Loading a data:
		-	List of supported spectra and file types
		-	How to load basic spectrum
		-	How to load spectrum from command line
		-	How to use the loader wizard [Provide quick tips or link to the Loader Wizard section]
		-	How to make a custom loader for unsupported data
	-	Exporting Spectra:
		-	As FITS?
		-	As ECSV 
	-	Workspaces and Plots:
		-	List all icons in on the toolbar and what they do
		-	How to create and remove new workspaces and plots 
	-	Using Regions:
		-	How to make and remove regions 
		-	Specify how the regions affect statistics and model fitting or link to the pages that do this
-	Unit Conversion section looks good.
-	Arithmetic Widget section looks good.
-	Loader Wizard section looks good.
	-	My personal openion is that this should be merged with a larger “how to load data section”
-	Improve Model fitting section by revising steps and adding screenshots.
-	Spectrum Statistics section looks good but we should indicate that the tabs on the side of specviz toggle b/w statistics and fitting.
-	Sections needed:
	-	Line labeling section. 
-	Proofread each page for typos. 
	-	On the index page (front page), remove the “SpecViz will soon include the ability to:” paragraph

@kassin @javerbukh

**Comment by robelgeda** [comment]
Also, quick link to current docs:
http://specviz.readthedocs.io

**Comment by astrofrog** [comment]
To install the latest-ish developer version with conda (built nightly):

```
conda install -c glueviz/label/dev specviz
```

I could potentially make the builds more frequent, and there might even be a way to trigger a build on a merge to master (but not sure)

**Comment by kassin** [comment]
main page
---------

"SpecViz is built on top of the SpecUtils Astropy-affiliated python library, providing a visual, interactive interface to the analysis capabilities in that library."
->
"SpecViz provides an interactive interface to the SpecUtils astropy-affiliated python library."
(Keep link to SpecUtils.)

remove "notably"

"data-analysis" -> "data analysis"

Add our high priority user-facing issues ONLY to the "soon include the ability to" list.

Installation
------------

Need a notice as to which version(s) of Python SpecViz works with and which it does not.

period after "detailed below"

The sentences starting with "distributed through... channel." are misplaced?

-Install via Anaconda-
Include nstructions to check if the user has Anaconda installed.

"then all you have to do to install SpecViz is simply type the..."
->
"then type the..."

"your install command" -> "the below install command"

channel -> <channel> in both the text and the command

"simply type the following" -> "type the following"

-Install via source-

"followint" -> "following"

-By using pip-
"Clone the SpecViz repository somewhere on your system, and install locally using pip."

Walk the user through how to do this (the instructions for cloning are below so they should be moved to before the "By using pip" section)

-PyQT bindings-

It is unclear if the user needs to install this and if so/not why?


Launching SpecViz
-----------------
no comments


Getting Started
---------------
Unclear what a "simple known spectrum is."

I suggest "test spectrum" and include a link to it.

"spectrum data file" -> "spectrum"

"you will be greeted with a plotted spectrum" -> "the spectrum should plot"

"exported spectrum file" -> "exported spectrum"


-Workspaces and Plots- (This should be its own page.)

Loaded data is added to the global data list for the entire workspace. A workspace is an instance of the SpecViz application. Several can be open at once and more can be added by selecting File > New Workspace from the menu bar. Workspaces are completely independent of one another, and the user can have as many workspaces are they wish.

**Comment by kassin** [comment]
-Workspaces and Plots- (This should be its own page.)

Robel to edit based on Glue docs explaining similar concepts.

-Using Regions- (This should be its own page.)

Interesting regions can be defined by adding an ROI to the plot. This is done via the Add Region button in the plot toolbar. Multiple regions can be used when defining e.g. areas the user might wish to use for model fitting.

A blue highlighted region is considered the _active_ region, and will in cases where it is imported (e.g. statistics), will be used to define the region used in the calculations.

->

A user may select single or multiple wavelength ranges of spectra for model fitting or measuring statistics.  This is done with the "Region of Interest" (or, ROI) selector via the Add Region button in the plot toolbar.  When a region is selected, it is highligted with a blue shading.  This region will be used for the statistics panel and the model fitting.


Currently, only one region can be the “selected” region. This will change in the future to allow a subset of plotted region to be defined as “selected”.

->

Currently one a single region can be selected.


Removing a region is accomplished by clicking the Remove Region button. Only the currently selected region is removed.

->

(move before the "Note")

To remove a region, click on the Remove Region button.

Unit Conversion
---------------

Unit Conversion uses the astropy.units module in order to convert spectral axis and flux units.

->

SpecViz allows the user to change the units on their spectra.  It uses the astropy.units module to do this.

(Why is the import astropy.units command here?)

The GUI comboboxes are then populated by potential_X_axis_units and potential_Y_axis_units, respectively. The user can then select any of those options - as well as a “Custom” option - and the changes will be reflected in the plot. If the user selects the “Custom” option, they can type in their own units, and if the units are accepted by astropy.units the unit conversion dialog box will allow the units to be parsed. However, “Custom” units are checked to see if the conversion is allowed before any conversion occurs and if False, the units will not be changed.

->

(Walk the user through the button clicks needed to change units.
I suggest removing all the developer lingo from this paragraph and
writing for an astronomer who does not use astropy or program in python.
You might want to include notes for the astropy user, but they should be at
the bottom of the page.  Currently astronomers would be scared off
from all the technical stuff at the top.)

Arithmetic Widget -> Arithmetic Editor (these two terms are used interchangeably, leading to confusion- just choose "Editor")
-----------------
The Arithmetic widget allows users to perform basic arithmetic on their spectra that are loaded into Specviz. The arithmetic editor accepts data loaded into the Specviz session and only outputs type specutils.Spectrum1D. This means the editor allows users to perform the same arithmatic operations that are available with specutils.Spectrum1D objects.

->

The Arithmetic Editor allows users to perform basic spectral arithmetic. Inputs are spectra loaded into the Specviz session.  Outputs are also spectra in SpecViz.

All spectra are specutils.Spectrum1D objects in astropy. Therefore, the Arithmetic Editor allows users to perform the  arithmatic operations that are incorporated into the specutils.Spectrum1D object.

Remove: "Launching Arithmetic Editor"  It doesn't need to be it's own sub-section.

**Comment by kassin** [comment]
Lets start by loading the arithmetic layer widget located on the tool bar. Upon clicking you will be prompted with the arithmetic dialog

->

To load the Arithmetic Editor, click on the ??? in the tool bar.  This dialog box should appear:

From this dialog you can add, edit or remove arithmetic items from the editor.  We will start by clicking the New Arithmetic Attribute button located in top left hand corner of the Editor dialog. Upon clicking you will be prompted with the editor dialog

->

We will walk you through an example of how to multiple a spectrum by a factor of 2.  Start by clicking the "New Arithmetic Attribute" button in the top left and the following dialog box will appear:

remove "Adding Arithmetic" section heading

Add:  Arithmetic is done using expressions entered into the large box.  Examples of operations are given at the bottom of the dialog box.  Let's choose a spectrum to operate on called "science spectrum" by choosing it from the first drop-doen menu and clicking "Insert." (UNSURE??)  Let's choose to operate on the wavelength axis bu choosing "wavelength" from the second drop-down menu and clicking "Insert."

(Walk the user through the rest of the steps in this manner... I'm not sure what they are.)

Loader Wizard
-------------
The loader wizard will assist users in generating a custom data loader. After working through the wizard a new loader python file can be saved to a directory of the user’s choice (default is the ~/.specutils directory). The written loader will then be automatically added to the specutils loader registry and accessible from the file open dialog.

->

The Loader Wizard assists with loading spectra into SpecViz.  For a given spectrum it creates a custom loader file which can then be used to load the spectrum (and spectra in the same format) without having to use the wizard. (Can you translate the last sentence into plain English?)

The Loader Wizard is invoked by clicking on the...  the below dialog box will display:


SpecViz uses astropy tables to read in spectra (specifically, astropy.table.Table.read).  Therefore, the syntax in the spectra to be read in should follow that used by astropy.table.Table.read, which is found here (add link).

(I find this paragraph really hard to follow.  Robel, can you take a shot at explaining the
Loader Wzard dialog box from scratch?)

Model Fitting
-------------
SpecViz utilizes Astropy Models and SpecUtils Fitting to fit models to its spectra. For example, you can fit one model to the continuum, another to an emission line of interest, and yet another to an absorption line.

Currently, the following models are available:

The models can be fitted with the following fitters:

->

SpecViz includes functionality to fit models to spectra.  It uses astropy's Models (??) and SpecUtils to do so.  The following models are available in SpecViz:


They can be fit with the following algorithms (from ??? in astropy):

(In this description, the use of the word "model" is non-standard.  Generally
a model is a function used to fit to data.  It is unclear what a "model data item"
is.  Rewrite this for a non-astropy user who simply wants to fit a Gaussian to an
emission line.  Perhaps walk the user through this use case using screen shots
to illustrate most steps?)

Spectrum Statistics -> Statistics
-------------------

The user can calculate statistics for a seleted region of a spectrum.  First, a "Region of Interest"
is selected (link to how to do this).  Once this is done, the statistics are automatically calculated
and displayed...

The following statistics are calculated.  They are from astropy (except for RMS).... (link to doc for each of these which are in astropy).

**Comment by robelgeda** [comment]
- Index:
	- [ ] SpecViz logo/banner needs to be larger above the side bar menu.

- Installation:
	- [ ] Can SpecViz run on python 2.x?
	- [ ] Include nstructions to check if the user has Anaconda installed.
	- [ ] How to clone specviz repo
	- [ ] -PyQT bindings- make unclear if the user needs to install this and if so/not why?

- Loading data:
	- [ ] List of supported spectra and file types
	- [ ] How to load basic spectrum
	- [ ] How to load spectrum from command line
	- [ ] How to use the loader wizard [Provide quick tips or link to the Loader Wizard section]
	- [ ] How to make a custom loader for unsupported data

- Exporting Spectra:
	- [ ] As ECSV
	- [ ] As FITS?

- Workspaces and Plots:
	- [ ] List all icons in on the toolbar and what they do
	- [ ] How to create and remove new workspaces and plots
	- [ ] How to make plot visible / how to hide plot
	- [ ] Plot units compatibility rules
	- [ ] How to select a data

- Using Regions:
	- [ ] How to make and remove regions
	- [ ] Specify how the regions affect statistics and model fitting or link to the pages that do this

- Model fitting:
	- [ ] Improve Model fitting section by revising steps and adding screenshots.

- Line labeling:
	- [ ] Make line labeling section.

- Operations
	- [ ] Make operations section (smoothing etc...)


- Susan's review:
	- [ ] Main page
	- [ ] Installation
	- [ ] Getting Started
	- [ ] Unit Conversion
	- [ ] Arithmetic Editor
	- [ ] Loader Wizard
	- [ ] Model Fitting
	- [ ] Statistics

**Comment by robelgeda** [comment]
# Progress on Susan's Review:

main page
---------

~~"SpecViz is built on top of the SpecUtils Astropy-affiliated python library, providing a visual, interactive interface to the analysis capabilities in that library."~~
->
~~"SpecViz provides an interactive interface to the SpecUtils astropy-affiliated python library."
(Keep link to SpecUtils.)~~
~~remove "notably"~~

~~"data-analysis" -> "data analysis"~~

Add our high priority user-facing issues ONLY to the "soon include the ability to" list.

Installation
------------

Need a notice as to which version(s) of Python SpecViz works with and which it does not.

period after "detailed below"

The sentences starting with "distributed through... channel." are misplaced?

-Install via Anaconda-
Include nstructions to check if the user has Anaconda installed.

"then all you have to do to install SpecViz is simply type the..."
->
"then type the..."

"your install command" -> "the below install command"

channel -> <channel> in both the text and the command

"simply type the following" -> "type the following"

-Install via source-

"followint" -> "following"

-By using pip-
"Clone the SpecViz repository somewhere on your system, and install locally using pip."

Walk the user through how to do this (the instructions for cloning are below so they should be moved to before the "By using pip" section)

-PyQT bindings-

It is unclear if the user needs to install this and if so/not why?


Launching SpecViz
-----------------
no comments


Getting Started
---------------
Unclear what a "simple known spectrum is."

I suggest "test spectrum" and include a link to it.

"spectrum data file" -> "spectrum"

"you will be greeted with a plotted spectrum" -> "the spectrum should plot"

"exported spectrum file" -> "exported spectrum"


-Workspaces and Plots- (This should be its own page.)

-Workspaces and Plots- (This should be its own page.)

Robel to edit based on Glue docs explaining similar concepts.

-Using Regions- (This should be its own page.)

Interesting regions can be defined by adding an ROI to the plot. This is done via the Add Region button in the plot toolbar. Multiple regions can be used when defining e.g. areas the user might wish to use for model fitting.

A blue highlighted region is considered the _active_ region, and will in cases where it is imported (e.g. statistics), will be used to define the region used in the calculations.

->

A user may select single or multiple wavelength ranges of spectra for model fitting or measuring statistics.  This is done with the "Region of Interest" (or, ROI) selector via the Add Region button in the plot toolbar.  When a region is selected, it is highligted with a blue shading.  This region will be used for the statistics panel and the model fitting.


Currently, only one region can be the “selected” region. This will change in the future to allow a subset of plotted region to be defined as “selected”.

->

Currently one a single region can be selected.


Removing a region is accomplished by clicking the Remove Region button. Only the currently selected region is removed.

->

(move before the "Note")

To remove a region, click on the Remove Region button.

Unit Conversion
---------------

Unit Conversion uses the astropy.units module in order to convert spectral axis and flux units.

->

SpecViz allows the user to change the units on their spectra.  It uses the astropy.units module to do this.

(Why is the import astropy.units command here?)

The GUI comboboxes are then populated by potential_X_axis_units and potential_Y_axis_units, respectively. The user can then select any of those options - as well as a “Custom” option - and the changes will be reflected in the plot. If the user selects the “Custom” option, they can type in their own units, and if the units are accepted by astropy.units the unit conversion dialog box will allow the units to be parsed. However, “Custom” units are checked to see if the conversion is allowed before any conversion occurs and if False, the units will not be changed.

->

(Walk the user through the button clicks needed to change units.
I suggest removing all the developer lingo from this paragraph and
writing for an astronomer who does not use astropy or program in python.
You might want to include notes for the astropy user, but they should be at
the bottom of the page.  Currently astronomers would be scared off
from all the technical stuff at the top.)

Arithmetic Widget -> Arithmetic Editor (these two terms are used interchangeably, leading to confusion- just choose "Editor")
-----------------
The Arithmetic widget allows users to perform basic arithmetic on their spectra that are loaded into Specviz. The arithmetic editor accepts data loaded into the Specviz session and only outputs type specutils.Spectrum1D. This means the editor allows users to perform the same arithmatic operations that are available with specutils.Spectrum1D objects.

->

The Arithmetic Editor allows users to perform basic spectral arithmetic. Inputs are spectra loaded into the Specviz session.  Outputs are also spectra in SpecViz.

All spectra are specutils.Spectrum1D objects in astropy. Therefore, the Arithmetic Editor allows users to perform the  arithmatic operations that are incorporated into the specutils.Spectrum1D object.

Remove: "Launching Arithmetic Editor"  It doesn't need to be it's own sub-section.

Lets start by loading the arithmetic layer widget located on the tool bar. Upon clicking you will be prompted with the arithmetic dialog

->

To load the Arithmetic Editor, click on the ??? in the tool bar.  This dialog box should appear:

From this dialog you can add, edit or remove arithmetic items from the editor.  We will start by clicking the New Arithmetic Attribute button located in top left hand corner of the Editor dialog. Upon clicking you will be prompted with the editor dialog

->

We will walk you through an example of how to multiple a spectrum by a factor of 2.  Start by clicking the "New Arithmetic Attribute" button in the top left and the following dialog box will appear:

remove "Adding Arithmetic" section heading

Add:  Arithmetic is done using expressions entered into the large box.  Examples of operations are given at the bottom of the dialog box.  Let's choose a spectrum to operate on called "science spectrum" by choosing it from the first drop-doen menu and clicking "Insert." (UNSURE??)  Let's choose to operate on the wavelength axis bu choosing "wavelength" from the second drop-down menu and clicking "Insert."

(Walk the user through the rest of the steps in this manner... I'm not sure what they are.)

Loader Wizard
-------------
The loader wizard will assist users in generating a custom data loader. After working through the wizard a new loader python file can be saved to a directory of the user’s choice (default is the ~/.specutils directory). The written loader will then be automatically added to the specutils loader registry and accessible from the file open dialog.

->

The Loader Wizard assists with loading spectra into SpecViz.  For a given spectrum it creates a custom loader file which can then be used to load the spectrum (and spectra in the same format) without having to use the wizard. (Can you translate the last sentence into plain English?)

The Loader Wizard is invoked by clicking on the...  the below dialog box will display:


SpecViz uses astropy tables to read in spectra (specifically, astropy.table.Table.read).  Therefore, the syntax in the spectra to be read in should follow that used by astropy.table.Table.read, which is found here (add link).

(I find this paragraph really hard to follow.  Robel, can you take a shot at explaining the
Loader Wzard dialog box from scratch?)

Model Fitting
-------------
SpecViz utilizes Astropy Models and SpecUtils Fitting to fit models to its spectra. For example, you can fit one model to the continuum, another to an emission line of interest, and yet another to an absorption line.

Currently, the following models are available:

The models can be fitted with the following fitters:

->

SpecViz includes functionality to fit models to spectra.  It uses astropy's Models (??) and SpecUtils to do so.  The following models are available in SpecViz:


They can be fit with the following algorithms (from ??? in astropy):

(In this description, the use of the word "model" is non-standard.  Generally
a model is a function used to fit to data.  It is unclear what a "model data item"
is.  Rewrite this for a non-astropy user who simply wants to fit a Gaussian to an
emission line.  Perhaps walk the user through this use case using screen shots
to illustrate most steps?)

Spectrum Statistics -> Statistics
-------------------

The user can calculate statistics for a seleted region of a spectrum.  First, a "Region of Interest"
is selected (link to how to do this).  Once this is done, the statistics are automatically calculated
and displayed...

The following statistics are calculated.  They are from astropy (except for RMS).... (link to doc for each of these which are in astropy).

**Comment by robelgeda** [comment]
@SaOgaz Suggested not hiding the loader wizard page. I have re-enabled it and will leave it as is for now.
